### DISCUSSION TITLE
What does the error message "transaction could not be decoded: unsupported transaction type" mean when attempting to broadcast a script on zkSync, and how can I resolve it?

### DISCUSSION THREAD

**Discussion by vasanthmarimuth** [Cyfrin/foundry-full-course-cu#2697] [Q&A]
Help me to resolve it. I am using a infura rpc url for zksync and using  zkSync Sepolia Testnet, the version I used is ^0.8.16.

**Comment by EngrPips** [comment]
Hello @vasanthmarimuth, As of now, the Foundry script does not work with `zksync`, so forget it till working with `zksync` becomes easier.
